tui displays node peers from environment

ffakenz opened the pull request:
<!-- Describe your change here -->

Closes #2044

By including the Environment of the running hydra-node in the Greetings message, clients can now use this information to display the expected peers the node should be connected with.

---

<!-- Consider each and tick it off one way or the other -->
* [x] CHANGELOG updated or not needed
* [x] Documentation updated or not needed
* [x] Haddocks updated or not needed
* [x] No new TODOs introduced or explained herafter

ffakenz committed:
add json instances for SigningKey HydraKey

ffakenz committed:
add environment to greetings

ffakenz committed:
draft new handling logic for server outputs

> now peers are shown in different colors depending on their status (connected/disconnected)

ffakenz committed:
fmt

v0d1ch commented on hydra-tx/src/Hydra/Tx/Crypto.hs:
```diff
@@ -187,6 +187,18 @@ instance HasTextEnvelope (VerificationKey HydraKey) where
+   where
+    deserialiseKey =
+      maybe
+        (fail "unable to deserialize SigningKey, wrong length")
```
I guess this can also fail for some reason other than wrong length?

ffakenz replied:
probably, I just reused the same error msg that's used in the VerificationKey instance

v0d1ch commented on hydra-node/json-schemas/api.yaml:
```diff
@@ -2788,6 +2791,45 @@ components:
       description: |
         Current hydra-node version.
 
+    DepositPeriod:
```
Why did you have to add `DepositPeriod` here? Seems unrelated

ffakenz replied:
I'm passing the full Environment to Greetings for potential future use, even though only `configuredPeers` is used for now.

ffakenz committed:
add adt arbitrary for greetings and rountrip golden spec

ffakenz committed:
introduce peer is unknown status

ffakenz committed:
minor fix to status coloring style

ffakenz committed:
fix: preserve peer statuses when handling final Greetings event

The peer list may already be partially populated from earlier events.
Now, we retain known statuses for existing peers and assign 'unknown' to newly configured ones.

ffakenz committed:
update CHANGELOG
